episodes/filters-and-thresholding.md

```diff
@@ -0,0 +1,743 @@
+cell sample image"}
+
+From this histogram, we can see two peaks - a larger one at low intensity, then 
+a smaller one at higher intensity. Looking at the image, it would make sense if 
+the lower intensity peak represented the background, with the higher one 
+representing the nuclei. We can verify this by adjusting the contrast limits in 
+the layer controls. If we move the left contrast limits node to the right of the 
+low intensity peak (around 8266), we can still clearly see the nuclei:
+
+![](fig/contrast-limit-8266-nuclei.png){alt="Left, fluorescence microscopy image 
+of cells. Right, histogram of the same image. Both with left contrast limit set 
+to 8266."}
+
+If we move the left contrast limits node to the right of the higher intensity 
+peak (around 28263), most of the nuclei disappear from the image:
+
+![](fig/contrast-limit-28263-nuclei.png){alt="Left, fluorescence microscopy 
+image of cells. Right, histogram of the same image. Both with left contrast 
+limit set to 28263."}
+
+Recall that you can set specific values for the contrast limits by right 
+clicking on the contrast limits slider in Napari.
+
+This demonstrates that the higher intensity peak is the one we are interested 
+in, and we should set our threshold accordingly.
+
+We can set this threshold with some commands in Napari's console:
+
+```python
+
+# Get the image data for the nuclei
+nuclei = viewer.layers["nuclei"].data
+
+# Create mask with a threshold of 8266
+mask = nuclei > 8266
+
+# Add mask to the Napari viewer
+viewer.add_labels(mask)
+
+```
+
+Note that we use `.add_labels()` here, rather than `.add_image()` which we used 
+in previous episodes. `.add_labels()` will ensure our mask is added as a 
+`Labels` layer, giving us access to all the annotation tools and settings for 
+segmentations (as covered in the [manual segmentation episode
+](quality-control-and-manual_segmentation.md)). `add_image()` would create a 
+standard `Image` layer, which is not what we want in this case.
+
+You should see a mask appear that highlights the nuclei in brown. If we set the 
+nuclei contrast limits back to normal (select 'nuclei' in the layer list, then 
+drag the left contrast limits node back to zero), then toggle on/off the mask 
+with the ![](
+https://raw.githubusercontent.com/napari/napari/main/napari/resources/icons/visibility.svg
+){alt="A screenshot of Napari's eye button" height='30px'} icon, you should see 
+that the brown areas match the nucleus boundaries reasonably well. They aren't 
+perfect though! The brown regions have a speckled appearance where some regions 
+inside nuclei aren't labelled and some areas in the background are incorrectly 
+labelled. We need to find a way to smooth out this nuclei mask.
+
+![](fig/threshold-mask.png){alt="Mask of nuclei (brown) overlaid on nuclei 
+image - created with manual thresholding"}
+
+::::::::::::::::::::::::::::::::::::: challenge 
+
+## Manual thresholds
```

Comment:
I've clarified at the start of this exercise what the main aim is now. The main focus is to get students practising choosing thresholds based on the image's histogram + setting them as greater than a value, less than a value or between two values (rather than working with noisy images in particular). I only added noise to this image to give each shape a wider peak in the histogram, rather than a spike at only one pixel value which would never happen in real life. Happy to add an additional exercise more focused on noise using sample images (as you mentioned in your comment) if you think it's necessary.